examples/platform/silabs/efr32/rs911x/rsi_if.c

```diff
@@ -469,12 +490,17 @@ static void wfx_rsi_do_join(void)
          */
         wfx_rsi.dev_state |= WFX_RSI_ST_STA_CONNECTING;
 
+        if ((status = rsi_wlan_register_callbacks(RSI_JOIN_FAIL_CB, wfx_rsi_join_fail_cb)) != RSI_SUCCESS)
+        {
+            WFX_RSI_LOG("%s: RSI callback register join failed with status: %02x", __func__, status);
+        }
+
         /* Try to connect Wifi with given Credentials
          * untill there is a success or maximum number of tries allowed
          */
-        while (++wfx_rsi.join_retries < WFX_RSI_CONFIG_MAX_JOIN)
+        while (true)
```

Comment:
True ?
and then wfx_rsi.join_retries++


and later you do                     
if (wfx_rsi.join_retries < MAX_JOIN_RETRIES_COUNT)
                    {
                        WFX_RSI_LOG("%s: Next attempt after 5 Sec", __func__);
                        vTaskDelay(pdMS_TO_TICKS(WLAN_RETRY_TIMER));
                    }
                    else
                        break;

i would say do a for loop then

Reply:
This API called either commissioning time or reconnection time. As mentioned in description while reconnection if AP failed to come online then after maximum retries device not able to connect with AP. To avoid this at telescopic interval (up-to max limit) device try to reconnect with AP and after max_limit_timer(time interval) device continuously trying to reconnect with AP at max_limit_timer(time interval) . 

```
while(true)
{
    retry++;
    if(asyn_connect_failed)
    {
        if(commissiomg_time)            //Comissioning time
        {
            if(Not_max_retry_reach)
            {
                delay(fixed_delay);
            }
            else
                break;
        }
        else                                        //Reconnection time
       {                          
            if(under_max_limit)
            {
                delay(telescopic_delay)
            }
            else{
                delay(max_limit_delay)
            }
        }
    }
    else
        break;
}
```